scheduler/core/eventsqueue/events.py

```diff
@@ -137,19 +139,24 @@ class WeatherChangeEvent(InterruptionEvent):
 class FaultEvent(InterruptionEvent):
     """
     Interruption that occurs when there is a fault in a resource.
-    TODO: Should we have one per Resource (probably), or one for multiple resources?
+    In OCS, this will likely be the site itself where the fault occurred.
     """
     affects: FrozenSet[Resource]
```

Comment:
It shouldn't matter, because while `WeatherClosureEvent` and `FaultEvent` have a shared abstract superclass parent, it doesn't contain `affects`, so in essence, what I'm saying is that `WeatherClosureEvent` uses a property to determine its `affects` and `FaultEvent` uses its member.

I think this will work fine, but if during testing, something happens, then we can modify the event hierarchy slightly.